### PR TITLE
Delete superseded phone number on add

### PR DESCRIPTION
**Pull request by buoyad**
On adding a phone number, check if we currently have a superseded item for that phone number. If it exists, delete it. This makes the yellow banner on account setting disappear. 

Also did some cleanup while looking around this code.
cc @keybase/y2ksquad

### CODE REVIEW

File: shared/actions/settings.tsx [hunk 1]
```diff
@@ -373,12 +363,16 @@ const loadSettings = async (
       (settings.emails || []).map(row => [row.email, Constants.makeEmailRow(row)])
     )
     const phoneMap: I.Map<string, Types.PhoneRow> = I.Map(
-      // Sort the superseded numbers first, so that if a number exxists in both
-      // superseded and non-superseded form, the non-superseded version ends up
-      // in the map.
-      (settings.phoneNumbers || [])
-        .sort(comparePhoneRows)
-        .map(row => [row.phoneNumber, Constants.toPhoneRow(row)])
+      (settings.phoneNumbers || []).reduce(
```
Comment (buoyad) [inline shared/actions/settings.tsx]: cc @jakob223 I changed this to a reduce rather than sorting

File: shared/settings/account/index.tsx [hunk 2]
```diff
@@ -143,7 +143,7 @@ const AccountSettings = (props: Props) => (
         <Kb.Banner color="yellow" onClose={props.onClearSupersededPhoneNumber}>
           <Kb.BannerParagraph
             bannerColor="yellow"
-            content={`Your unverified phone number ${
+            content={`Your phone number ${
```
Comment (buoyad) [inline shared/settings/account/index.tsx]: cc @heronhaye / @zapu I noticed this wording was wrong after having a verified number become superseded. Is that a bug? It makes sense to me that that'd work.

**Comment by buoyad** [comment]
@jakob223 does the whole PR look good or just what I cc'd you on?

**Comment by jakob223** [comment]
Whole PR

**Pull request merged**